### PR TITLE
Fixed permissions bits after extracting the charm's content to analyze (CRAFT-429)

### PR DESCRIPTION
**Pull request by facundobatista**

### CODE REVIEW

File: tests/commands/test_analyze.py [hunk 1]
```diff
@@ -76,6 +76,29 @@ def fake_analyze(passed_config, passed_basedir, *, override_ignore_config):
     assert fake_analyze_called
 
 
+@pytest.mark.parametrize("modebits", [0o777, 0o750, 0o444])
+def test_expanded_charm_permissions(config, tmp_path, monkeypatch, modebits):
+    """Check that the expanded charm keep original permissions."""
```
Comment (sergiusens) [inline tests/commands/test_analyze.py:81]:
```suggestion
    """Check that the expanded charm keeps original permissions."""
```

File: charmcraft/commands/analyze.py [hunk 2]
```diff
@@ -61,23 +61,37 @@ def fill_parser(self, parser):
...
+        tmpdir = pathlib.Path(tempfile.mkdtemp())
         try:
-            zf = zipfile.ZipFile(filepath)
-            zf.extractall(path=tmpdir)
+            zf = zipfile.ZipFile(str(filepath))
+            zf.extractall(path=str(tmpdir))
         except Exception as exc:
             raise CommandError(
-                "Cannot open the indicated charm file {!r}: {!r}.".format(filepath, exc)
+                "Cannot open the indicated charm file {!r}: {!r}.".format(str(filepath), exc)
```
Comment (sergiusens) [inline charmcraft/commands/analyze.py:72]:
```suggestion
                "Cannot open charm file {!r}: {!r}.".format(str(filepath), exc)
```

Comment (sergiusens) [reply charmcraft/commands/analyze.py:72]: Mayb f-string :-)

**Review by sergiusens — APPROVED** [review]

File: charmcraft/commands/analyze.py [hunk 3]
```diff
@@ -61,23 +61,37 @@ def fill_parser(self, parser):
...
             )
 
+        # fix permissions as extractall does not keep them (see bpo15795)
```
Comment (cjp256) [inline charmcraft/commands/analyze.py:75]: googling bpo15795 found nothing :-/

```suggestion
        # fix permissions as extractall does not keep them (see https://bugs.python.org/issue15795)
```

Very interesting.

**Review by cjp256 — APPROVED** [review]
With suggested changes, :+1: Thanks!

File: tests/commands/test_analyze.py (hunk 1 above)
Comment (facundobatista) [reply tests/commands/test_analyze.py:81]: thanks

File: charmcraft/commands/analyze.py (hunk 2 above)
Comment (facundobatista) [reply charmcraft/commands/analyze.py]: Improved.

File: charmcraft/commands/analyze.py (hunk 3 above)
Comment (facundobatista) [reply charmcraft/commands/analyze.py]: Ack.

**Pull request merged**